### ISSUE TITLE
Broken Extended CSS rule with `:upward()`

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/DeadDomainsLinter#12]
`spac1.ru,spcs.social#?#a[title="Реклама"]:upward(1)`
`:upward(1)` => `:upward()`

![image](https://github.com/AdguardTeam/DeadDomainsLinter/assets/8361299/b7c5d8a3-a56c-411b-9080-e4b2f078a5c3)

**Comment by ameshkov** [comment]
Looks like a bug in AGTree btw

CC @scripthunter7

**Comment by scripthunter7** [comment]
@ameshkov Yes, this is an AGTree bug. We have a custom generate function for CSS Tree nodes in AGTree v1.x.x (because by default CSS Tree omit all spaces, but we want generate some formatted CSS), and `Number` node is missed in [that function](https://github.com/AdguardTeam/tsurlfilter/blob/e27098b4d328a910a2bf3e2b1649a8675fd89e1f/packages/agtree/src/utils/csstree.ts#L689)
In AGTree v2, we don't use CSS Tree directly. So parsing happens correctly, but custom generator has a bug.
Should I fix v1.x.x for now, until we release v2?
